parts/k8s/kubernetesbase.tmpl

```diff
@@ -28,6 +28,43 @@
     {{template "k8s/kubernetesmastervars.tmpl" .}}
   },
   "resources": [
+  {{if IsUsingCosmos }}
```

Comment:
I still don't think this belongs in kubernetesbase.tmpl which just puts all the nested resources together. This is definitely a master resource. Having to put it in two different files (VMAS and VMSS) is unfortunate I agree and we have tried to reconcile all the files that were duplicated when master VMSS was added but the resource template was too different to easily merge into one. In the future we should think of other ways to avoid this. Maybe add another master resource file for everything that is common to VMAS and VMSS. But in any case, master resource definitions do not belong in this base template that is used by AKS as well.

Reply:
Though i disagree. I moved it to the sub files as requested.